src/plugins/dashboard/public/dashboard_app/_dashboard_app_strings.ts

```diff
@@ -154,6 +154,26 @@ export const shareModalStrings = {
       defaultMessage:
         'One or more panels on this dashboard have changed. Before you generate a snapshot, save the dashboard.',
     }),
+  getEmbedShareWarning: () =>
+    i18n.translate('dashboard.embedShare.draftWarning', {
+      defaultMessage:
+        'This dashboard has unsaved changes. Consider saving your dashboard before copying the embed link.',
+    }),
+  getDraftSharePanelChangesWarning: (hasSessionStorageEnabled: boolean) =>
+    hasSessionStorageEnabled
+      ? i18n.translate('dashboard.snapshotShare.panelChangesWarning', {
+          defaultMessage:
+            'One or more panels on this dashboard have changed. Before generating a link, save the dashboard.',
+        })
+      : i18n.translate('dashboard.snapshotShare.panelChangesWarningNoSessionStorage', {
+          defaultMessage:
+            'One or more panels on this dashboard have changed, resolving the link for this draft state requires session storage to be enabled, consider saving changes to the dashboard.',
+        }),
+  getDraftShareWarning: () =>
+    i18n.translate('dashboard.snapshotShare.draftWarning', {
+      defaultMessage:
+        'This dashboard has unsaved changes. Consider saving your dashboard before generating the link.',
+    }),
```

Comment:
@florent-leborgne @wajihaparvez Do you mind reviewing all of the new copy here? 🙇